crates/subgraph/src/performance/order_performance.rs

```diff
@@ -0,0 +1,765 @@
+                input: output.token.clone(),
+                output: input.token.clone(),
+            };
+            // if not same io token and ratio map doesnt already include them
+            if input.token != output.token
+                && !(pair_ratio_map.contains_key(&pair_as_key)
+                    || pair_ratio_map.contains_key(&inverse_pair_as_key))
+            {
+                // find this pairs(io or oi) latest trade from list of order's
+                // trades, to calculate the pair ratio (in amount/out amount) and
+                // its inverse from the latest trade that involes these 2 tokens.
+                let ratio = trades
+                    .iter()
+                    .find(|v| {
+                        (v.input_vault_balance_change.vault.token == input.token
+                            && v.output_vault_balance_change.vault.token == output.token)
+                            || (v.output_vault_balance_change.vault.token == input.token
+                                && v.input_vault_balance_change.vault.token == output.token)
+                    })
+                    .and_then(|latest_trade| {
+                        // convert input and output amounts to 18 decimals point
+                        // and then calculate the pair ratio
+                        let ratio = match latest_trade.ratio() {
+                            Ok(v) => Some(v),
+                            Err(e) => {
+                                if let PerformanceError::DivByZero = e {
+                                    Some(U256::MAX)
+                                } else {
+                                    None
+                                }
+                            }
+                        };
+                        let inverse_ratio = match latest_trade.inverse_ratio() {
+                            Ok(v) => Some(v),
+                            Err(e) => {
+                                if let PerformanceError::DivByZero = e {
+                                    Some(U256::MAX)
+                                } else {
+                                    None
+                                }
+                            }
+                        };
+                        ratio.zip(inverse_ratio).map(|(ratio, inverse_ratio)| {
+                            if latest_trade.input_vault_balance_change.vault.token == input.token {
+                                [ratio, inverse_ratio]
+                            } else {
+                                [inverse_ratio, ratio]
+                            }
+                        })
+                    });
+
+                // io
+                pair_ratio_map.insert(pair_as_key, ratio.map(|v| v[0]));
+                // oi
+                pair_ratio_map.insert(inverse_pair_as_key, ratio.map(|v| v[1]));
+            }
+        }
+    }
+
+    pair_ratio_map
+}
+
+fn combine(new_val: (U256, bool), old_val: (U256, bool)) -> (U256, bool) {
```

Comment:
test?

Reply:
added, fn has been renamed to accumulate()